src/HealthCheckResultTask.cs

```diff
@@ -10,3 +10,15 @@ internal static class HealthCheckResultTask
     /// </summary>
     public static readonly Task<HealthCheckResult> Healthy = Task.FromResult(HealthCheckResult.Healthy());
 }
+
+internal static class StringListExtensions
+{
+    public static HealthCheckResult GetHealthState(this List<string> instance, HealthCheckContext context)
+    {
+        if (instance.Count == 0)
```

Comment:
```suggestion
    public static HealthCheckResult GetHealthState(this List<string>? instance, HealthCheckContext context)
    {
        if (instance?.Count ?? 0 == 0)
```

Reply:
Doesn't it look strange if the GetHealthState method works on `null` objects? Writing the suggested code will allow the following code:

```
List<string>? t = null;
if (t is null)
{
    t.GetHealthState(context);
}
```

Code is valid and works but this is weird.

Reply:
> Doesn't it look strange if the GetHealthState method works on null objects? 

In our case - no, because `null` means the _no errors_.